File: dotnet-client/src/SolvexClient.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

public class SolvexClient {
    private readonly HttpClient _client;
    private readonly string _apiKey;

    public SolvexClient(string apiKey) {
        _apiKey = apiKey;
        _client = new HttpClient();
    }

    public async Task<string> GetLeaderboardAsync() {
        _client.DefaultRequestHeaders.Add("Authorization", "Bearer " + _apiKey);
        var response = await _client.GetStringAsync("https://api.solvex.ai/leaderboard");
        return response;
    }
}

File: swift-client/src/SolvexClient.swift
import Foundation

class SolvexClient {
    let apiKey: String
    let baseUrl = "https://api.solvex.ai"

    init(apiKey: String) {
        self.apiKey = apiKey
    }

    func getLeaderboard(completion: @escaping (String?) -> Void) {
        guard let url = URL(string: "\(baseUrl)/leaderboard") else { return }
        var request = URLRequest(url: url)
        request.setValue("Bearer \(apiKey)", forHTTPHeaderField: "Authorization")

        let task = URLSession.shared.dataTask(with: request) { data, _, _ in
            let response = data.flatMap { String(data: $0, encoding: .utf8) }
            completion(response)
        }
        task.resume()
    }
}
